os/binfmt/binfmt_loadbinary.c

```diff
@@ -136,27 +136,34 @@ int load_binary(int binary_idx, FAR const char *filename, load_attr_t *load_attr
 		/* Initialize the binary structure */
 
 		bin->filename = filename;
+
 		if (load_attr) {
-			bin->exports = NULL;
-			bin->nexports = 0;
-			bin->filelen = load_attr->bin_size;
-			bin->offset = load_attr->offset;
-			bin->stacksize = load_attr->stack_size;
-			bin->priority = load_attr->priority;
-			bin->compression_type = load_attr->compression_type;
-			bin->binary_idx = binary_idx;
-			bin->bin_ver = load_attr->bin_ver;
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+			if (binary_idx == BM_CMNLIB_IDX) {
+				bin->islibrary = true;
+				bin->filelen = load_attr->bin_size;
+				bin->offset = load_attr->offset;
+				bin->bin_ver = load_attr->bin_ver;
```

Comment:
Is it okay not to set the compression type?

Reply:
The compression is not supported for common binary now.